Devnot.Mentor.Api/Repositories/Interfaces/IMentorApplicationsRepository.cs

```diff
@@ -7,5 +10,8 @@ public interface IMentorApplicationsRepository : IRepository<MentorApplications>
     {
         Task<bool> IsExistsByUserId(int mentorId, int menteeId);
         Task<MentorApplications> Get(int mentorId, int menteeId);
+        Task<IEnumerable<MentorApplications>> GetForMentees(Expression<Func<MentorApplications, bool>> predicate);
+        Task<IEnumerable<MentorApplications>> GetForMentors(Expression<Func<MentorApplications, bool>> predicate);
```

Comment:
Bu kısımlarda direkt `Expression` kullanılması çok uygun değil gibi. Servis içerisinde metotu birçok yer kullanıyor ve tam olarak nasıl kullanıldığı hakkında bir fikrimiz de olmuyor.

Örnek vermek gerekirse;

```csharp
GetForMentees(x => x.MenteeId == mentee.Id && x.Status != MentorApplicationStatus.Approved.ToInt())
GetForMentees(x => x.MenteeId == mentee.Id)
GetForMentees(x => x.MentorId == mentor.Id)
```

gibi birçok senaryo üretilebiliyor Servis sınıfları içerisinde. Bu da metotun neyi neden yaptığını çok da açıklamıyor açıkçası. Bu kısımlar metot overload olabilir ya da çok parametre alan bir yapı varsa da model olarak gönderilip Repository kısımlarında LINQ sorgularına denk düşürülebilir.

Böyle daha sağlıklı olacağı kanaatindeyim.

Reply:
Repository katmanında metotların çoğalacağını düşündüğüm için bunu tercih etmiştim, öyleyse tamamdır bununla alakalı değişiklikleri de yapıyorum.